src/applications/vaos/tests/e2e/workflows/referrals-workflow/referral-api-errors.cypress.spec.js

```diff
@@ -365,5 +365,74 @@ describe('VAOS Referral API Error Handling', () => {
         completeReferral.assertApiError();
       });
     });
+
+    it('should display an error when appointment remains in proposed state and times out', () => {
+      // Mock appointment details to always return proposed status (never transitions to booked)
+      const proposedAppointmentResponse = new MockReferralAppointmentDetailsResponse(
+        {
+          appointmentId,
+          typeOfCare: 'OPTOMETRY',
+          providerName: 'Dr. Bones',
+          organizationName: 'Meridian Health',
+          status: 'proposed',
+          success: true,
+        },
+      ).toJSON();
+
+      // Use the simpler mock - app will handle polling internally
+      mockAppointmentDetailsApi({
+        id: appointmentId,
+        response: proposedAppointmentResponse,
+      });
+
+      // Use cy.clock() to control time and speed up the polling timeout test
+      cy.clock(new Date('2025-06-02T12:00:00Z'), ['Date']);
```

Comment:
I was trying something like this at some point and having trouble getting it to work. Nice to see a working example.